reference_system/include/reference_system/number_cruncher.hpp

```diff
@@ -18,34 +18,23 @@
 #include <cmath>
 #include <vector>
 
-std::vector<uint64_t> number_cruncher(const std::chrono::nanoseconds & timeout)
+uint64_t number_cruncher(const uint64_t maximum_number)
 {
-  // cause heavy CPU load by finding some primes
-  std::vector<uint64_t> primes{2};
-  // allocate memory beforehand so that we put only load on the CPU
-  primes.reserve(1024 * 1024);
+  uint64_t number_of_primes = 0;
   bool has_timeout_occurred = false;
-  auto current_time = [] {
-      return std::chrono::duration_cast<std::chrono::nanoseconds>(
-        std::chrono::system_clock::now().time_since_epoch())
-             .count();
-    };
-  int64_t start_time = current_time();
-  for (uint64_t i = 3; !has_timeout_occurred; ++i) {
+  for (uint64_t i = 3; i < maximum_number; ++i) {
     uint64_t rootOfI = static_cast<uint64_t>(std::sqrt(i));
-    for (auto p : primes) {
-      if (current_time() - start_time > timeout.count()) {
-        has_timeout_occurred = true;
-      }
-      if (p > rootOfI) {
-        primes.emplace_back(i);
-        break;
-      } else if (i % p == 0) {
+    bool is_prime = true;
+    for (uint64_t n = 2; n < rootOfI; ++n) {
+      if (i % n == 0) {
         break;
```

Comment:
@elfenpiff shouldn't you set `is_prime` to `false` here then?
```suggestion
        is_prime = false;
        break;
```

Reply:
otherwise the later check for `is_prime` will always result in `true` i think, right?